#451 Added chocolatey package support for Functions.

martinpinto opened the pull request:
Added a new package for chocolatey for the latest Functions version.

martinpinto committed:
Added chocolatey package support for Functions.

seiflotfy committed:
Merge branch 'master' into fn_chocolatey

denismakogon commented on fn/chocolatey/functions.nuspec:
```diff
@@ -0,0 +1,21 @@
+    <projectUrl>https://github.com/iron-io/functions/</projectUrl>
+    <!--<iconUrl>http://cdn.rawgit.com/__REPLACE_YOUR_REPO__/master/icons/functions.png</iconUrl>-->
+    <copyright>2017 Functions Iron.io</copyright>
+    <tags>serverless faas docker lambda</tags>
```
IFAIK we'd like to keep ourselves for being associated with Serverless (see https://serverless.com/). So it'll be better to have next tags:
`functions faas docker containers` (as an alternative - include COE there).

treeder replied:
I'd say keep serverless and add functions. We do want to be associated with the term serverless as that's what everyone is referring to it as for the time being.

martinpinto replied:
ok done. Kept `serverless` and added `functions`.

denismakogon commented on fn/chocolatey/functions.nuspec:
```diff
@@ -0,0 +1,21 @@
+    
+    <title>Iron.io Functions (Install)</title>
+    <authors>Iron.io Inc.</authors>
+    <projectUrl>https://github.com/iron-io/functions/</projectUrl>
```
trailing slash? just remove slash right after `functions`

martinpinto replied:
ok done. Removed both the trailing slash from the `<projectUrl>` and `<packageSourceUrl>`.

treeder committed:
Merge branch 'master' into fn_chocolatey

treeder commented on fn/chocolatey/tools/chocolateyinstall.ps1:
```diff
@@ -0,0 +1,24 @@
+
+$packageName= 'functions'
+$toolsDir   = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
+$url        = 'https://github.com/iron-io/functions/releases/download/0.2.25/fn.exe'
```
Could this URL always grab the latest release?  Similar to `install.sh` ?

martinpinto replied:
Unfortunately this URL is bound to that version. We might need to adjust scripts to auto increase the version. Updated version to 0.2.36.

treeder replied:
Yes, we need to make the scripts update the version on release, same way as we do for git, docker, cli install, etc. 

How does this get submitted to Chocolatey?  We'll want to make sure that's automated too.

martinpinto committed:
Fix #451. Added review changes.

denismakogon committed:
Merge branch 'master' into fn_chocolatey

denismakogon committed:
Merge branch 'master' into fn_chocolatey

seiflotfy committed:
Fix lambda import (#605)

* Fix lambda import

* more updates

* more changes

Commit:
functions: 0.2.55 release [skip ci]

denismakogon committed:
Add idle_timeout to routes API (#603)

* Add inactivity_timeout to routes API

Closes: #544

* Fix failing datastore tests

* Rename inactivity_timeout to idle_timeout

* Update swagger doc

* Update hot fn doc

* Fix json tags

* Add function timeouts docs

* Rewording

Commit:
functions: 0.2.56 release [skip ci]

pedronasser committed:
fix: error unmarshal empty app config (#609)

pedronasser committed:
Solving postgres marshal/unmarshal issue (#610)

* Solving postgres marshal/unmarshal issue

Postgres datastore was not marshaling the App config during its insert, that behavior was resulting in issues when fetching the App and the datastore couldn't unmarshal the config.

The same issue was probably happening with the Route's headers in some situations.

This commit's idea is to always try to marshal configs and headers when inserting/updating Apps or Routes. But in Apps and Routes get methods, if the  config/headers unmarshal fails, it returns an empty config/headers.

* fix one more unmarshal case

* returning error when unmarshaling non-empty

Commit:
functions: 0.2.57 release [skip ci]

martinpinto committed:
Added automatic update to release version for chocolatey scripts.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/iron-io/functions?pullRequest=564) <br/>All committers have signed the CLA.
